grails-app/views/dashboard/megamenu.gsp

```diff
@@ -101,14 +102,33 @@
   $(document).ready(function () {
     const path = window.location.pathname
     const menuConfigValues = $(".menu-config-value").toArray();
-    const matchingSection = menuConfigValues.find(it => it.value.includes(path))
-    // Assign active-section class to matched section
-    if (matchingSection) {
-      const matchingMenuSection = $("#" + matchingSection?.name).get(0);
-      const matchingMenuSectionCollapsable = $("#" + matchingSection?.name + "-collapsed").get(0);
-      if (matchingMenuSection) matchingMenuSection.classList.add('active-section');
-      if (matchingMenuSectionCollapsable) matchingMenuSectionCollapsable.classList.add('active-section');
-    }
+    const urlPartsWithSectionConfig = "${(grailsApplication.config.openboxes.menuSectionsUrlParts as JSON)}";
```

Comment:
I feel like it could be better to return it from the backend, from the `def megamenu` just adding another property to return after `menu`, `configurationSection`, `megamenuConfig`:
```
menuSectionsUrlParts: grailsApplication.config.openboxes.menuSectionsUrlParts as JSON
```

and to create a hidden field with id/class, that would allow you to access it from the js script side

Reply:
@kchelstowski Yup, good suggestion